lib/ts/recipe/emailpassword/index.ts

```diff
@@ -20,27 +20,140 @@ import RecipeUserId from "../../recipeUserId";
 import { DEFAULT_TENANT_ID } from "../multitenancy/constants";
 import { getPasswordResetLink } from "./utils";
 import { getRequestFromUserContext, getUser } from "../..";
+import { getUserContext } from "../../utils";
+import { SessionContainerInterface } from "../session/types";
+import { User } from "../../types";
 
 export default class Wrapper {
     static init = Recipe.init;
 
     static Error = SuperTokensError;
 
-    static signUp(tenantId: string, email: string, password: string, userContext?: any) {
+    static signUp(
+        tenantId: string,
+        email: string,
+        password: string,
+        session?: undefined,
```

Comment:
I think we should have this, it'd make it hard to "re-implement" our APIs with session based linking if someone wants to do it in a custom API.

Reply:
We could implement it, but then the types would be fairly messy for the last two parameters, since removing this param (only in some cases) would move the userContext param which now can appear in two places.